src/main/App.tsx

```diff
@@ -137,25 +138,32 @@ export default class App extends React.Component<
     let contextIRI = urlParams.get("workspace");
     if (contextIRI && isURL(contextIRI)) {
       contextIRI = decodeURIComponent(contextIRI);
-      if (contextIRI.includes("/diagram-")) {
-        this.loadVocabularies(contextIRI, AppSettings.contextEndpoint);
-      } else this.loadVocabularies(contextIRI, AppSettings.contextEndpoint);
+      this.loadVocabularies(contextIRI, AppSettings.contextEndpoint);
     } else {
       this.handleStatus(
         false,
-        Locale[AppSettings.viewLanguage].pleaseReload,
+        Locale[AppSettings.interfaceLanguage].pleaseReload,
         true
       );
     }
   }
 
+  handleChangeInterfaceLanguage(languageCode: string) {
```

Comment:
Yes, I believe we have had such a conversation before. I agree; I found this especially frustrating at the time of adding many features to various panels (such as this) and considered adding Redux to fix it. The problem is that there is always another more important bug or feature... I'll look into ways of introducing Redux gradually.

Reply:
Maybe it would be worth looking into this project - it's a simple Redux alternative that can be adopted gradually easily. https://github.com/lostpebble/pullstate